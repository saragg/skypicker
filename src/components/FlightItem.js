import React from 'react';
import './FlightItem.css';
import PropTypes from 'prop-types';
import FontIcon from 'material-ui/FontIcon'
import {Card, CardText} from 'material-ui/Card';
import FlagIconFactory from 'react-flag-icon-css';
import Route from './FlightItemRoute';

const FlagIcon = FlagIconFactory(React, { useCssModules: false })

const MoreInfoItem = (props) => {
  return (
    <div className="list-item">
      <FontIcon style={{"color": "#666"}} className={props.icon} />
      <span className="text"> {props.text} </span>
      {props.value}
    </div>
  )
}

const FlighItem = (props) => {

  const moreInfo = [
    {
      icon: "fa fa-clock-o",
      text: "Departure time",
      value: props.item.fly_duration
    },{
      icon: "fa fa-clock-o",
      text: "Return time",
      value: props.item.return_duration
    },{
      icon: "fa fa-plane",
      text: "Planes",
      value: props.item.route.length
    },{
      icon: "fa fa-bed",
      text: "Nights in destination",
      value: props.item.nightsInDest
    }
  ]

  return(
    <Card style={{marginBottom: "20px"}}>
      <CardText>
        <div className="flight-box">
          <div>
            <div className="title">
              <div>
                {props.item.cityFrom}({props.item.flyFrom})
                <FlagIcon className='flag-icon'
                  code={props.item.countryFrom.code.toLowerCase()}
                />
              </div>
              <div>
                <FontIcon style={{color: "#666", margin: "0 16px", fontSize: "12px"}}
                  className="fa fa-arrows-h"
                />
              </div>
              <div>
                {props.item.cityTo}({props.item.flyTo})
                <FlagIcon className='flag-icon'
                  code={props.item.countryTo.code.toLowerCase()}
                />
              </div>
            </div>
            <div>
              {props.item.route.map((route) => <Route key={route.id} route={route} />)}
            </div>
          </div>
          <div className="more-info">
            <div className="list">
              {moreInfo.map((info, i) =>
                <MoreInfoItem key={i} icon={info.icon} text={info.text} value={info.value} />)
              }
            </div>
          </div>
          <div className="price">
            {props.item.price}&euro;
          </div>
        </div>
      </CardText>
    </Card>
  )
}

FlighItem.propTypes = {
  item: PropTypes.object.isRequired,
};

export default FlighItem;