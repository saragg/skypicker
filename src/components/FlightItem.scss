.flight-box {
    display: flex;
    -webkit-justify-content: space-between;
    justify-content: space-between;
    @media (max-width: 900px) {
        flex-direction: column;
    }
    .title {
        font-size: 12px;
        color: #333;
        font-weight: 500;
        display: flex;
        -webkit-justify-content: space-between;
        justify-content: space-between;
        @media (max-width: 900px) {
            display: none;
        }
    }
    .price {
        font-size: 24px;
        font-weight: 100;
        color: limegreen;
        @media (max-width: 900px) {
            text-align: center;
        }
    }
    .route {
        color: #999;
        display: flex;
        -webkit-justify-content: space-between;
        justify-content: space-between;
        margin-top: 8px;
        .time {
            font-size: 16px;
            color: #333;
        }
    }
    .more-info {
        .list {
            list-style-type: none;
            padding: 0;
            display: flex;
            flex-direction: column;
            .list-item {
                margin-bottom: 8px;
            }
            span {
                margin-left: 8px;
            }
            @media (max-width: 900px) {
                margin-top: 16px;
                flex-direction: row;
                span {
                    margin: 0 8px;
                    display: none;
                }
            }
        }

    }
    .flag-icon {
      padding: 0;
      margin-left: 8px;
    }
}





