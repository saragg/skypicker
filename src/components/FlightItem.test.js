import React from 'react';
import FlightItem from './FlightItem';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

test('Renders correctly', () => {
  const item = {
    cityFrom: "Barcelona",
    flyFrom: "Tokio",
    countryFrom: {
      code: 'ES',
    },
    countryTo: {
      code: 'JP',
    },
    route: [],
    fly_duration: "18h 30m",
    return_duration: "20h 45m",
    nightsInDest: 3,
    price: 170,
  }

  const tree = renderer.create(
    <MuiThemeProvider>
      <FlightItem item={item}/>
    </MuiThemeProvider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});