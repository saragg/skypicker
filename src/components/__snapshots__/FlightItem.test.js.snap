// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "marginBottom": "20px",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  >
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      <div
        className="flight-box"
      >
        <div>
          <div
            className="title"
          >
            <div>
              Barcelona
              (
              Tokio
              )
              <span
                className="flag-icon flag-icon-es"
              />
            </div>
            <div>
              <span
                className="fa fa-arrows-h"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "#666",
                    "display": "inline-block",
                    "fontSize": "12px",
                    "margin": "0 16px",
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              />
            </div>
            <div>
              (
              )
              <span
                className="flag-icon flag-icon-jp"
              />
            </div>
          </div>
          <div />
        </div>
        <div
          className="more-info"
        >
          <div
            className="list"
          >
            <div
              className="list-item"
            >
              <span
                className="fa fa-clock-o"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "#666",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              />
              <span
                className="text"
              >
                 
                Departure time
                 
              </span>
              18h 30m
            </div>
            <div
              className="list-item"
            >
              <span
                className="fa fa-clock-o"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "#666",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              />
              <span
                className="text"
              >
                 
                Return time
                 
              </span>
              20h 45m
            </div>
            <div
              className="list-item"
            >
              <span
                className="fa fa-plane"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "#666",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              />
              <span
                className="text"
              >
                 
                Planes
                 
              </span>
              0
            </div>
            <div
              className="list-item"
            >
              <span
                className="fa fa-bed"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "#666",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              />
              <span
                className="text"
              >
                 
                Nights in destination
                 
              </span>
              3
            </div>
          </div>
        </div>
        <div
          className="price"
        >
          170
          â‚¬
        </div>
      </div>
    </div>
  </div>
</div>
`;
